src/main/java/seedu/notus/command/EditEventCommand.java

```diff
@@ -18,23 +35,256 @@
+                    warningSignalRecurrence = true;
+                    break;
+                }
+                assert (endRecurrenceDate != null);
+                newEvent = new DailyEvent(eventTitle, eventStartDateTime, eventIsToRemind, endRecurrenceDate,
+                        event.getReminderPeriods());
+                break;
+            case RecurringEvent.WEEKLY_RECURRENCE_TYPE:
+                if (event instanceof WeeklyEvent) {
+                    warningSignalRecurrence = true;
+                    break;
+                }
+                assert (endRecurrenceDate != null);
+                newEvent = new WeeklyEvent(eventTitle, eventStartDateTime, eventIsToRemind, endRecurrenceDate,
+                        event.getReminderPeriods());
+                break;
+            case RecurringEvent.MONTHLY_RECURRENCE_TYPE:
+                if (event instanceof MonthlyEvent) {
+                    warningSignalRecurrence = true;
+                    break;
+                }
+                assert (endRecurrenceDate != null);
+                newEvent = new MonthlyEvent(eventTitle, eventStartDateTime, eventIsToRemind, endRecurrenceDate,
+                        event.getReminderPeriods());
+                break;
+            case RecurringEvent.YEARLY_RECURRENCE_TYPE:
+                if (event instanceof YearlyEvent) {
+                    warningSignalRecurrence = true;
+                    break;
+                }
+                assert (endRecurrenceDate != null);
+                newEvent = new YearlyEvent(eventTitle, eventStartDateTime, eventIsToRemind, endRecurrenceDate,
+                        event.getReminderPeriods());
+                break;
+            default:
+                // Should not hit here.
+                warningSignalRecurrence = true;
+                break;
+            }
+            if (warningSignalRecurrence) {
+                results.add(COMMAND_WARNING_RECURRENCE_MESSAGE);
+            } else {
+                results.add(COMMAND_SUCCESSFUL_RECURRENCE_MESSAGE);
+                timetable.setEvent(index, newEvent);
+            }
+
+        }
+
+        // Edit endRecurrenceDate here.
+        if (endRecurrenceDate != null) {
+            Event currEvent = timetable.getEvent(index);
+            if (currEvent instanceof RecurringEvent) {
+                results.add(COMMAND_SUCCESSFUL_RECURRENCE_DATE_MESSAGE);
+                ((RecurringEvent) currEvent).setEndRecurrenceDate(endRecurrenceDate);
+            } else {
+                // Throw warnings
+                results.add(COMMAND_WARNING_RECURRENCE_ON_NON_RECURRENCE_MESSAGE);
+            }
+        }
+
+
+        if (results.size() == 1) {
+            results.add(COMMAND_UNSUCCESSFUL_MESSAGE);
+        } else {
+            try {
+                storageManager.saveTimetable(timetable);
+            } catch (IOException exception) {
+                results.add(FILE_WRITE_UNSUCCESSFUL_MESSAGE);
```

Comment:
This line may not be necessary since you return in the next line. But other than that looks good to me

Reply:
I was thinking of doing this because it may have edited the data but failed the saving